mne/io/ctf/ctf.py

```diff
@@ -149,13 +150,13 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
         with open(self._filenames[fi], 'rb') as fid:
             for bi in range(len(r_lims)):
                 samp_offset = (bi + trial_start_idx) * si['res4_nsamp']
-                n_read = min(si['n_samp'] - samp_offset, si['block_size'])
+                n_read = min(si['n_samp_tot'] - samp_offset, si['block_size'])
```

Comment:
Both the Python and C versions can omit samples from the end. We do it differently from the C tools to handle a user's use case, see:

https://github.com/mne-tools/mne-python/pull/2712

I suspect this fix will break that use case, but I'm not sure. We really do need to add some test files to figure it out.